common/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/network/GenericNetworkQuery.kt

```diff
@@ -31,3 +30,57 @@ GetPasswordTokenInteractor) : PaginationOffsetLimitQuery(identifier ?: "$offset-
 class OAuthPasswordObjectQuery<T>(value: T, val getPasswordTokenInteractor: GetPasswordTokenInteractor) : ObjectQuery<T>(value),
     OAuthClientQuery
 
+/**
+ * Implement the interface and override getPasswordTokenInteractor when you need oAuth authorization.
+ */
+interface GenericOAuthQuery {
+  val getPasswordTokenInteractor: GetPasswordTokenInteractor
+}
+
+/**
+ * Base class.
+ * In case you want to customize the cache key, provide a key string.
+ */
+open class GenericNetworkQuery(
+    val path: String,
+    val params: List<Pair<String, String>> = emptyList(),
+    val headers: List<Pair<String, String>> = emptyList(),
+    key: String? = null) : KeyQuery(key ?: path)
+
```

Comment:
I would like to make all attributes open so we can do something like this when extending the class:
`  override val headers: List<Pair<String, String>>
    get() {
      // Network related code that we only need to be used if we are going to network
    }`